pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -158,7 +158,10 @@ func prepareLocalProfileServerMap(localServerURL string, getconfigCtx *getconfig
 			}
 			if localProfileServerIP != nil {
 				//check if defined IP of localServer is equals with allocated IP of app
-				if ulStatus.AllocatedIPAddr == localProfileServerIP.String() {
+				if len(ulStatus.AllocatedIPAddr) == 0 {
+					continue
+				}
+				if ulStatus.AllocatedIPAddr[0] == localProfileServerIP.String() {
```

Comment:
Shouldn't this compare all of them e.g., by defining some hasAllocatedIPAddr() function?

Reply:
split the allocated addresses field two (one per ip version).